apisix/init.lua

```diff
@@ -324,6 +324,28 @@ function _M.http_access_phase()
 
     core.ctx.set_vars_meta(api_ctx)
 
+    local uri = api_ctx.var.uri
+    if local_conf.apisix and local_conf.apisix.delete_uri_tail_slash then
+        if str_byte(uri, #uri) == str_byte("/") then
+            api_ctx.var.uri = str_sub(api_ctx.var.uri, 1, #uri - 1)
+            core.log.info("remove the end of uri '/', current uri: ",
+                          api_ctx.var.uri)
+        end
+    end
+
+    if router.api.has_route_not_under_apisix() or
```

Comment:
Maybe we can add a configuration in the `config.yaml` for it?

Reply:
@spacewander  do you mean a switch for it ?

Reply:
Yes

Reply:
@spacewander 
I thought about it again. There shouldn't have a switch here. If skip the `if` branch and continue execution, a 404 will occur because the route does not exist.

And I think if need to configure the plugin for the internal api, it is more appropriate to configure a route and then configure the plugin instead of using the global rule. 

What do you think ? Looking forward to your opinion. Thanks.

Reply:
Better to provide a switch to execute the global rule before matching internal API or don't execute the global rule. It is not skipped the internal API matched.

> And I think if need to configure the plugin for the internal api, it is more appropriate to configure a route and then configure the plugin instead of using the global rule.

We can't configure a plugin for internal API, this is the real problem.

Reply:
@spacewander  Thanks for for the explanation. I know that.., emm, will update later.

Reply:
OK. We can wait for it.